File: WaterContainer.java
public class WaterContainer {
    public int maxArea(int[] height) {
        int max = Integer.MIN_VALUE;
        int start = 0;
        int end = height.length - 1;
        while (start < end) {
            int area = Math.min(height[start], height[end]) * (end - start);
            if (area > max) {
                max = area;
            }
            if (height[start] < height[end]) {
                start++;
            } else {
                end--;
            }
        }
        return max;
    }
}

File: WaterContainer.py
from typing import List
class Solution:
    def maxArea(self, height: List[int]) -> int:
        maxA, left, right = 0, 0, len(height) - 1
        while left < right:
            maxA = max(maxA, min(height[left], height[right]) * (right - left))
            if height[left] < height[right]:
                left += 1
            else:
                right -= 1
        return maxA   